types/src/on_chain_config.rs

```diff
@@ -7,9 +7,77 @@ use crate::{
     language_storage::{StructTag, TypeTag},
     transaction::SCRIPT_HASH_LENGTH,
 };
+use anyhow::{format_err, Result};
 use libra_crypto::HashValue;
 use move_core_types::identifier::Identifier;
-use serde::{Deserialize, Serialize};
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use std::{collections::HashMap, sync::Arc};
+
+/// To register an on-chain config in Rust:
+/// 1. Implement the `OnChainConfig` trait for the Rust representation of the config
+/// 2. Add the config's `ConfigID` to `ON_CHAIN_CONFIG_REGISTRY`
+
+#[derive(Copy, Clone, Eq, Hash, PartialEq)]
+pub struct ConfigID(&'static str);
+
+impl ConfigID {
+    pub fn access_path(self) -> AccessPath {
+        access_path_for_config(Identifier::new(self.0).expect("failed to get Identifier"))
+    }
+}
+
+pub const ON_CHAIN_CONFIG_REGISTRY: &[ConfigID] = &[VMPublishingOption::CONFIG_ID];
```

Comment:
worth adding some comments about this e.g. it'd panic if the config is not initialized etc